clarity/src/vm/functions/mod.rs

```diff
@@ -664,20 +664,38 @@ fn special_asserts(
 
 pub fn handle_binding_list<F, E>(
     bindings: &[SymbolicExpression],
+    binding_error_type: SyntaxBindingErrorType,
     mut handler: F,
 ) -> std::result::Result<(), E>
 where
     F: FnMut(&ClarityName, &SymbolicExpression) -> std::result::Result<(), E>,
     E: From<CheckErrors>,
```

Comment:
Okay -- this is where I think my prior suggestion becomes problematic. This function returns `E: From<CheckErrors>` error types, which would be a problem if we're trying to use the `CheckError` to handle the symbolic expression. But the usage of this makes me even more skeptical of including portions of the AST in the error struct -- `handle_binding_list` also generates runtime errors (runtime-error wrapped CheckErrors). I don't think runtime errors should be cloning the AST. But anyways, I think a maybe cleaner thing to do here would be to update this function's signature:

```suggestion
where
    F: FnMut(&ClarityName, &SymbolicExpression) -> std::result::Result<(), E>,
    E: From<(SyntaxBindingError, &SymbolicExpression)>,
```

That conveys (correctly) that the handler only returns BindingErrors, so E just needs to know how to convert from those, and then we'd define the `From<(SyntaxBindingError, &SymbolicExpression)>` impls explicitly -- that way this function no longer sometime implicitly relies on the `From<CheckErrors>` implementation for runtime errors.

Reply:
Alternatively, we force the caller to supply a `convert_fn`:

```
pub fn handle_binding_list<F, E>(
    bindings: &[SymbolicExpression],
    binding_error_type: SyntaxBindingErrorType,
    on_err: G, 
    mut handler: F,
) -> std::result::Result<(), E>
where
    G: Fn(SyntaxBindingError, &SymbolicExpression) -> E,
    F: FnMut(&ClarityName, &SymbolicExpression) -> std::result::Result<(), E> {
```